crates/risc0-bonsai/src/host.rs

```diff
@@ -39,242 +33,63 @@ pub struct RecoveredBonsaiSession {
-                                error!(err);
-                                Err(backoff::Error::permanent(err))
-                            }
-                        }
-                    }
-                }
-            },
-        )
-    };
-}
-
 /// A [`Risc0BonsaiHost`] stores a binary to execute in the Risc0 VM and prove in the Risc0 Bonsai API.
 #[derive(Clone)]
 pub struct Risc0BonsaiHost<'a> {
     elf: &'a [u8],
     env: Vec<u8>,
     assumptions: Vec<AssumptionReceipt>,
     image_id: Digest,
-    client: Option<Client>,
-    ledger_db: LedgerDB,
+    _ledger_db: LedgerDB,
 }
 
 impl<'a> Risc0BonsaiHost<'a> {
     /// Create a new Risc0Host to prove the given binary.
-    pub fn new(elf: &'a [u8], api_url: String, api_key: String, ledger_db: LedgerDB) -> Self {
+    pub fn new(elf: &'a [u8], ledger_db: LedgerDB) -> Self {
         // Compute the image_id, then upload the ELF with the image_id as its key.
         // handle error
         let image_id = compute_image_id(elf).unwrap();
 
         tracing::trace!("Calculated image id: {:?}", image_id.as_words());
 
-        // handle error
-        let client = if !api_url.is_empty() && !api_key.is_empty() {
-            tracing::debug!("Uploading image with id: {}", image_id);
-            let elf = elf.to_vec();
-            thread::spawn(move || {
-                let client = Client::from_parts(api_url, api_key, risc0_zkvm::VERSION)
-                    .expect("Failed to create Bonsai client; qed");
-
-                client
-                    .upload_img(hex::encode(image_id).as_str(), elf)
-                    .expect("Failed to upload image; qed");
-
-                tracing::debug!("Image uploaded");
-
-                Some(client)
-            })
-            .join()
-            .unwrap()
-        } else {
-            None
-        };
+        match std::env::var("RISC0_PROVER") {
+            Ok(prover) => match prover.as_str() {
+                "bonsai" => {
+                    if std::env::var("BONSAI_API_URL").is_err()
+                        || std::env::var("BONSAI_API_KEY").is_err()
+                    {
+                        panic!("Bonsai API URL and API key must be set when RISC0_PROVER is set to bonsai");
+                    }
+                }
+                "local" => {}
+                "ipc" => {
+                    if std::env::var("RISC0_SERVER_PATH").is_err() {
+                        panic!("RISC0_SERVER_PATH must be set when RISC0_PROVER is set to ipc");
+                    }
+                }
+                _ => {
+                    panic!("Invalid prover specified: {}", prover);
+                }
+            },
+            Err(_) => {
+                debug!("No prover specified.");
+
+                if std::env::var("BONSAI_API_URL").is_ok()
+                    && std::env::var("BONSAI_API_KEY").is_ok()
+                {
+                    panic!(
```

Comment:
Can we not assume intent instead of panicking?

Reply:
I wanted to be very explicit so that we don't run into a case where we accidentally run some mode until it fails on proving